internal/internal.go

```diff
@@ -510,9 +516,20 @@ func getParsedPath(req *ext_authz.CheckRequest) []interface{} {
 	return parsedPathInterface
 }
 
-func getParsedBody(req *ext_authz.CheckRequest) map[string]interface{} {
+func getParsedBody(req *ext_authz.CheckRequest) (map[string]interface{}, error) {
 	body := req.GetAttributes().GetRequest().GetHttp().GetBody()
+	headers := req.GetAttributes().GetRequest().GetHttp().GetHeaders()
+
 	var data map[string]interface{}
-	util.Unmarshal([]byte(body), &data)
-	return data
+
+	if val, ok := headers["content-type"]; ok {
+		if val == "application/json" {
```

Comment:
We should probably check if the content-type _contains_ application/json instead of strict equality because if there are parameters then this will fail.

Reply:
Done.